### PR TITLE
feat(change-case): support unicode stuff

### LINKED ISSUE #4841: [change-case] not working correctly with Latin characters
### Extension

https://www.raycast.com/es183923/change-case

### Description

When you use the extension with Latin characters like `ó` or `'` they do not appear in the extension resulting in words with missing characters. I see that also commas`,` if they come after a Latin character, are not included.

### LINKED ISSUE #7420: [Change Case] Converting to sentence case removes special characters.
### Extension

https://www.raycast.com/es183923/change-case

### Description

Converting to sentence case removes special characters.

### Steps To Reproduce

Run the command on text containing comas, apostrophes, etc.

### Current Behaviour

_No response_

### Expected Behaviour

_No response_

### LINKED ISSUE #7452: [Change Case] Do not remove non-alphanumeric characters from converted text
### Extension

https://www.raycast.com/es183923/change-case

### Description

Currently, all non-alphanumeric characters (e.g. punctuation, symbols) are removed from the text, which makes it less than useful for converting English text. Please provide an option to preserve those characters.

### Who will benefit from this feature?

Everyone who wants to use this very useful extension for making case in English text consistent, without losing punctuation or other special characters.

### Anything else?

Thanks for the useful extension! I use it very frequently.

### PR DESCRIPTION
**Pull request by erics118**
## Description

updated the code a lot
now works with unicode stuff

issues regarding unicode stuff:
Fixes #4841 
Fixes #7452 
Fixes #7420 

## Screencast

todo

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/change-case/tsconfig.json
Comment (erics118) [inline extensions/change-case/tsconfig.json:1]: changes, as in deviations from the template extension

**Comment by erics118** [comment]
CI (or rather, the compiler) suggests adding `"type": "module"` to `package.json`, but then `ray dev` will cause it to crash with:
```
21:38:28.399 
Error: require() of ES Module /Users/eric/.config/raycast/extensions/change-case/change-case.js from /Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js not supported.
change-case.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename change-case.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/eric/.config/raycast/extensions/change-case/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).


change-case.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename change-case.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/eric/.config/raycast/extensions/change-case/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
Hq:index.js:23:34787
```

**Comment by pernielsentikaer** [comment]
Hi @erics118 👋 

Why did you change `tsconfig.json`, is there a specific reason for that?

**Comment by erics118** [comment]
I changed it in an another attempt to fix the compiler errors

**Comment by pernielsentikaer** [comment]
We don't support ESM commands right now, as we call `require` when initializing a command. It's something that we want to look into, but I can't give you any ETA regarding this.

**Comment by erics118** [comment]
@pernielsentikaer I fixed the issue with ESM-only packages.

I cloned the change-case repo locally, generated the dist files, and copied them over. Probably not the best way, but it seems to be the simplest solution

**Review by pernielsentikaer — APPROVED** [review]
Let's do like that until we hopefully support it 🙂

**Pull request merged**